src/main/java/com/intrafind/log4shell/Log4shellUpdate.java

```diff
@@ -146,6 +186,76 @@ private static void handleIfOldLog4j(Path path, Map<Path, String> toReplace, Lis
     }
   }
 
+  @SuppressWarnings("ConstantConditions")
+  private static void handleIfLog4j1(Path path, List<Path> toDelete, Map<Path, String> toAdd, Map<Path, String> toAppend, Map<Path, Map<String, String>> toReplaceInFile, Map<Path, String> toReplace) {
+    if (LOG4J1_PATTERN.asPredicate().test(path.getFileName().toString())) {
+      final Path basePath = path.getParent().getParent();
+      final Path[] slf4jImpls;
+      try {
+        slf4jImpls = Files.list(basePath.resolve("lib")).filter(file -> file.getFileName().toString().matches("slf4j-log4j12-.*\\.jar")).toArray(Path[]::new);
+      } catch (IOException e) {
+        System.err.println("Could not determine SLF4J implementation for " + basePath + ". Cannot replace " + path + " Exception: " + e.getMessage());
+        return;
+      }
+      if (slf4jImpls.length > 1) {
+        System.err.println("Could not determine SLF4J implementation for " + basePath + ". Cannot replace " + path);
+        return;
+      } else if (slf4jImpls.length == 1) {
+        toReplace.put(basePath.resolve("lib").resolve(slf4jImpls[0]), "log4j-slf4j-impl-2.17.0.jar");
+      }
+      if (isIntrafindService(path)) {
+        toDelete.add(basePath.resolve("log4j.properties"));
+        toAdd.put(basePath.resolve("log4j2.xml"), "log4j2.xml");
+        toAppend.put(basePath.resolve("conf/wrapper.conf"), LOG4J2_WRAPPER_SETTINGS);
+      } else if (isIntrafindWebapp(path)) {
+        toDelete.add(basePath.resolve("classes/log4j.properties"));
+        toAdd.put(basePath.resolve("classes/log4j2.xml"), "log4j2.xml");
+      } else if (isIntrafindApp(path)) {
+        String[] startScripts = basePath.resolve(IS_WINDOWS ? "bat" : "bin").toFile().list((dir, name) -> name.startsWith("start_"));
```

Comment:
You can either work with limit too or use findFirst if we do not need the information that we have more than 1.
It would potentially have a problem though if more than one startscript is found

Reply:
ah, forget it, it is not a stream here